File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/lab/binary_plus.cpp
// binary_plus.cpp

#include "binary.h"
#include "binary_plus.h"
class expression;
#include "expression_visitor.h"

binary_plus::binary_plus(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void binary_plus::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/lab/BinaryPlus.java
// BinaryPlus.java

public class BinaryPlus extends Binary
{
	public BinaryPlus(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}
